lib/framework/action.dart

```diff
@@ -201,7 +201,44 @@ class FileUploadAction extends EnsembleAction {
   String? uploadUrl;
 }
 
+class WalletConnectAction extends EnsembleAction {
+  WalletConnectAction({
+    required this.wcProjectId,
+    required this.appMetaData,
+    this.id,
+    this.onComplete,
+    this.onError,
+  });
+
+  String wcProjectId;
+  AppMetaData appMetaData;
+  String? id;
+  EnsembleAction? onComplete;
+  EnsembleAction? onError;
+
+}
+
+class AppMetaData {
+  String name;
+  String description;
+  String url;
+  String iconUrl;
+
+  AppMetaData({
+    required this.name,
```

Comment:
What kind of errors do you get if these required fields are not specified? If the errors are not shown in the Editor, try throwing an LanguageError